src/components/ListItemIcon/ListItemIcon.vue

```diff
@@ -93,64 +92,101 @@ import Avatar from '../Avatar'
 import Highlight from '../Highlight'
 import { userStatus } from '../../mixins'
 
+const margin = 6
+
 export default {
-	name: 'AvatarSelectOption',
+	name: 'ListItemIcon',
+
 	components: {
 		Avatar,
 		Highlight,
 	},
 	mixins: [userStatus],
+
 	props: {
 		/**
-		 * Secondary optional line
+		 * Default first line text
 		 */
-		desc: {
+		title: {
 			type: String,
-			default: '',
+			required: true,
 		},
+
 		/**
-		 * Default text
+		 * Secondary optional line
 		 */
-		displayName: {
+		subtitle: {
 			type: String,
-			required: true,
+			default: '',
 		},
+
 		/**
 		 * Icon class to be displayed at the end of the component
 		 */
 		icon: {
 			type: String,
 			default: '',
 		},
+
 		/**
-		 * See the [Avatar](#Avatar) user prop
+		 * Search within the highlight of title/subtitle
 		 */
-		user: {
+		search: {
 			type: String,
 			default: '',
 		},
+
 		/**
-		 * The status of the user
+		 * Set a size in px for the height
 		 */
-		status: {
-			type: Object,
+		size: {
+			type: Number,
+			default: 32,
+		},
+
+		/**
+		 * See the [Avatar](#Avatar) displayName prop
+		 * Fallback to title
+		 */
+		displayName: {
+			type: String,
 			default: null,
 		},
 		/**
 		 * See the [Avatar](#Avatar) isNoUser prop
+		 * Enable/disable the UserStatus fetching
 		 */
 		isNoUser: {
 			type: Boolean,
 			default: false,
 		},
+	},
 
-		search: {
-			type: String,
-			default: '',
+	data() {
+		return {
+			margin,
+		}
+	},
+
+	computed: {
+		hasIcon() {
+			return this.icon !== ''
+		},
+
+		hasSlot() {
+			return !!this.$slots.default
+		},
+		cssVars() {
```

Comment:
I've never seen this trick. Does the `--height` translate to a `height` in css?

Reply:
No, it's registering variables at the component level and can be accessed directly in the css with var(--xxxx). It is a nice trick to share variables between js and css cleanly.